r/tests/testthat/test-marker-heatmap.R

```diff
@@ -26,14 +26,14 @@ test_that("Marker Heatmap returns appropiate format", {
 
   res <- runMarkerHeatmap(req, data)
 
-  expect_equal(names(res), c("rawExpression", "truncatedExpression"))
+  expect_equal(names(res[[1]]), c("rawExpression", "truncatedExpression"))
 
   # number of rows is number of cells
-  expect_equal(ncol(data), nrow(res$rawExpression))
+  expect_equal(ncol(data), length(res[[1]]$rawExpression$expression))
```

Comment:
is there a way to avoid this `res[[1]]`? Looks very fragile. I saw it in other places too. What is `res[[1]]`? What is inside `res`?

Reply:
Res is a list, and we are accessing the first element in the list